source/common/upstream/upstream_impl.h

```diff
@@ -957,7 +959,7 @@ class ClusterInfoImpl : public ClusterInfo,
       lb_maglev_config_;
   const std::unique_ptr<const envoy::config::cluster::v3::Cluster::OriginalDstLbConfig>
       lb_original_dst_config_;
-  absl::optional<envoy::config::core::v3::TypedExtensionConfig> upstream_config_;
+  std::unique_ptr<envoy::config::core::v3::TypedExtensionConfig> upstream_config_;
```

Comment:
nit: const

Reply:
Can this also be `const TypedExtensionConfig`?

Reply:
After discussing this IRL it seems that this is non-const because of the tests, and I think that keeping it non-const is ok for now.